chia/plotting/manager.py

```diff
@@ -56,7 +56,11 @@ def __init__(
         self.no_key_filenames = set()
         self.farmer_public_keys = []
         self.pool_public_keys = []
-        self.cache = Cache(self.root_path.resolve() / "cache" / "plot_manager.dat")
+        # Since `compression_level` property was added to Cache structure,
+        # previous cache file formats needs to be reset
+        # When user downgrades harvester, it looks 'plot_manager.dat` while
+        # latest harvester reads/writes 'plot_manager_v2.dat`
+        self.cache = Cache(self.root_path.resolve() / "cache" / "plot_manager_v2.dat")
```

Comment:
I investigated diff between `chiapos@1.0.11` (pinned in `chia-blockchain@main`) and `chiapos@1.0.12b12` (pinned in `chia-blockchain@long_lived/compression`).
https://github.com/Chia-Network/chiapos/compare/1.0.11...1.0.12b13

![image](https://github.com/Chia-Network/chia-blockchain/assets/84098616/1b1450cb-cd84-4ca0-b818-1f6e6b55ba9d)

It seems new `chiapos` can serialize/deserialize both old and new versions of `DiskProver` binary.
But it doesn't mean old `chiapos` can deserialize binary created by new `chiapos`.


>The caching which is imo not ok. This should do an update of the cache and maybe even store the new cache in a separate file to keep the cache for downgrades since this changes the protocol and if there are downgrades for some reasons for big farms its just nice to have the cache available.

I feel your past suggestion above makes sense. When a user downgrade `chia-blockchain`, harvester would keep crashing because of cache deserialization failure.
I think `"plot_manager_v2.dat"` is nice and not messing up `cache` folder so much.

However, as you suggested, `compression_level` is not need in `CacheEntry` and `DiskCacheEntry`. I'll remove the property from both classes.